kernel/arch/dreamcast/include/dc/syscalls.h

```diff
@@ -0,0 +1,395 @@
+    \ingroup system_calls
+
+    This function returns a flashrom partition to all 1's, so that it may 
+    be rewritten.
+
+    \warning
+    ALL data in the entire partition will be lost.
+
+    \param  pos             The offset from the start of the flashrom you 
+                            want to delete.
+
+    \retval 0               On success.
+    \retval -1              On failure.
+
+    \sa syscall_flashrom_read(), syscall_flashrom_write()
+*/
+int syscall_flashrom_delete(uint32_t pos);
+
+/** \brief   Initialize the GDROM drive.
+    \ingroup system_calls
+
+    This function initializes the GDROM drive. Should be called before any 
+    commands are sent.
+*/
+void syscall_gdrom_init(void);
+
+/** \brief   Reset the GDROM drive.
+    \ingroup system_calls
+
+    This function resets the GDROM drive.
+*/
+void syscall_gdrom_reset(void);
+
+/** \brief   Checks the GDROM drive status.
+    \ingroup system_calls
+
+    This function retrieves the general condition of the GDROM drive. It 
+    populates a provided array with two elements. The first element 
+    indicates the current drive status, and the second element identifies 
+    the type of disk inserted (if any).
+
+    \param  status          The pointer to two 32-bit integers to 
+                            receive extended status information.
+
+    \return                 0 on success, or non-zero on
+                            failure.
+*/
+int syscall_gdrom_check_drive(void *status);
+
+/** \brief   Send a command to the GDROM command queue.
+    \ingroup system_calls
+
+    This function sends a command to the GDROM queue.
+
+    \note
+    Call syscall_gdrom_run_commands() to run queued commands.
+
+    \param  cmd             The command code (see CMD_* in cdrom.h).
+    \param  params          The pointer to parameter block for the command, 
+                            can be NULL if the command does not take 
+                            parameters.
+
+    \return                 The request id (>=0) on success, or <0 error
+                            code on failure.
+
+    \sa syscall_gdrom_check_command(), syscall_gdrom_run_commands()
+*/
+int syscall_gdrom_send_command(uint8_t cmd, void *params);
+
+/** \brief   Check status of queued command for the GDROM.
+    \ingroup system_calls
+
+    This function checks if a queued command has completed.
+
+    \param  id              The request id (>=0).
+    \param  status          The pointer to four 32-bit integers to 
+                            receive status information.
+
+    \retval 0               Request not found.
+    \retval 1               Request is still being processed.
+    \retval 2               Request completed successfully.
+    \retval 3               Stream type command is in progress.
+    \retval 4               GD syscalls are busy.
+
+    \sa syscall_gdrom_send_command(), syscall_gdrom_run_commands()
+*/
+int syscall_gdrom_check_command(uint32_t id, void *status);
```

Comment:
```suggestion
int syscall_gdrom_check_command(uint32_t id, int32_t status[4]);
```
That'd be a "pointer to four 32-bit integers."